Error when publishing release

```shell
npm notice Publishing to https://registry.npmjs.org/ with tag latest and public access
npm error code E404
npm error 404 Not Found - PUT https://registry.npmjs.org/@boykaframework%2fboyka-cli - Not found
npm error 404
npm error 404  '@boykaframework/boyka-cli@0.8.0' is not in this registry.
npm error 404
npm error 404 Note that you can also install from a
npm error 404 tarball, folder, http url, or git url.

npm error A complete log of this run can be found in: /home/runner/.npm/_logs/2024-07-20T09_52_16_347Z-debug-0.log

 ELIFECYCLE  Command failed with exit code 1.
 ELIFECYCLE  Command failed with exit code 1.
Error: Process completed with exit code 1.
```

--- a/.github/workflows/publish.yml
+++ b/.github/workflows/publish.yml
@@ -1,4 +1,4 @@
-name: Publish to GitHub
+name: Publish to NPM and GitHub
 
 on:
   workflow_dispatch:
@@ -21,19 +21,28 @@ env:
   NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
 
 jobs:
-  prepare:
+  publish:
     runs-on: ubuntu-latest
-
-    outputs:
-      new-version: ${{ steps.version.outputs.version }}
-      old-version: ${{ steps.old_version.outputs.previous_version }}
-      changelog: ${{ steps.changelog.outputs.changes }}
 
     steps:
       - name: Check out Git repository
         uses: actions/checkout@v4
         with:
           fetch-depth: 0
+          token: ${{ env.PUSH_TOKEN }}
+
+      - name: Import GPG key
+        uses: crazy-max/ghaction-import-gpg@v6
+        with:
+          gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
+          passphrase: ${{ secrets.GPG_PASSPHRASE }}
+          git_user_signingkey: true
+          git_commit_gpgsign: true
+
+      - name: git config
+        run: |
+          git config user.name "${GITHUB_ACTOR}"
+          git config user.email "${GITHUB_ACTOR}@users.noreply.github.com"
 
       - uses: pnpm/action-setup@v4
 
@@ -42,112 +51,12 @@ jobs:
         with:
           node-version: 18
           cache: 'pnpm'
-          registry-url: 'https://registry.npmjs.org'
+
+      - name: NPM Setup
+        run: pnpm config set //registry.npmjs.org/:_authToken $NPM_TOKEN
 
       - name: Install dependencies
         run: pnpm install && pnpm build
 
-      - name: Get the previous version
-        id: old_version
-        run: |
-          export OLD_VERSION=$(node -pe "require('./package.json').version")
-          echo "previous_version=${OLD_VERSION}" >> $GITHUB_OUTPUT
-
-      - name: Update the release version and publish to NPM
-        run: pnpm release:${{ inputs.release-type }}
-
-      - name: Get the new version
-        id: version
-        run: |
-          export NEW_VERSION=$(node -pe "require('./package.json').version")
-          echo "version=$NEW_VERSION" >> $GITHUB_OUTPUT
-
-      - name: Generate Change log
-        if: ${{ steps.version.outputs.version != null }}
-        id: changelog
-        run: |
-          export CHANGELOG=$(pnpm --silent changelog --from ${{ steps.old_version.outputs.previous_version }} --next-version ${{ steps.version.outputs.version }})
-          delimiter="$(openssl rand -hex 8)"
-          echo "changes<<${delimiter}" >> "${GITHUB_OUTPUT}"
-          echo "$CHANGELOG" >> "${GITHUB_OUTPUT}"
-          echo "${delimiter}" >> "${GITHUB_OUTPUT}"
-
-      - name: Print Changelog
-        run: echo "${{ steps.changelog.outputs.changes }}"
-
-      - name: Upload updated version related files to artifacts
-        uses: actions/upload-artifact@v4
-        with:
-          name: version-files-${{ github.run_id }}
-          retention-days: 1
-          path: |
-            ${{ github.workspace }}/lerna.json
-            ${{ github.workspace }}/package.json
-
-  push-pom:
-    runs-on: ubuntu-latest
-    needs:
-      - prepare
-
-    steps:
-      - name: Check out Git repository
-        uses: actions/checkout@v4
-        with:
-          token: ${{ env.PUSH_TOKEN }}
-
-      - name: Download Version files
-        uses: actions/download-artifact@v4
-        with:
-          name: version-files-${{ github.run_id }}
-
-      - name: Update Changelog file for v${{ needs.prepare.outputs.new-version }}
-        run: |
-          echo "${{ needs.prepare.outputs.changelog }}" > CHANGELOG.md
-
-      - name: Import GPG key
-        uses: crazy-max/ghaction-import-gpg@v6
-        with:
-          gpg_private_key: ${{ env.GPG_PRIVATE_KEY }}
-          passphrase: ${{ env.GPG_PASSPHRASE }}
-          git_user_signingkey: true
-          git_commit_gpgsign: true
-
-      - name: Push updated version files
-        uses: EndBug/add-and-commit@v9
-        with:
-          message: |
-            chore(Internal): :bookmark: released v${{ needs.prepare.outputs.new-version }} to NPM
-          push: true
-          default_author: user_info
-          commit: --no-verify
-
-  github:
-    runs-on: ubuntu-latest
-    needs:
-      - prepare
-      - push-pom
-
-    steps:
-      - name: Check out Git repository
-        uses: actions/checkout@v4
-        with:
-          fetch-depth: 0
-
-      - name: Download Version files
-        uses: actions/download-artifact@v4
-        with:
-          name: version-files-${{ github.run_id }}
-
-      - name: Create GitHub Release
-        uses: softprops/action-gh-release@v2
-        env:
-          GITHUB_TOKEN: ${{ env.PUSH_TOKEN }}
-          GITHUB_REPOSITORY: ${{ github.repository }}
-        with:
-          tag_name: ${{ needs.prepare.outputs.new-version }}
-          name: Version ${{ needs.prepare.outputs.new-version }}
-          prerelease: false
-          draft: false
-          body: ${{ needs.prepare.outputs.changelog }}
-          discussion_category_name: Announcements
-          generate_release_notes: false
+      - name: Release
+        run: pnpm release.${{ github.event.inputs.release-type }}

--- a/.release-it.json
+++ b/.release-it.json
@@ -1,16 +1,27 @@
 {
   "$schema": "https://unpkg.com/release-it@17/schema/release-it.json",
   "plugins": {
-    "release-it-pnpm": {
-      "disableRelease": true
+    "@release-it-plugins/lerna-changelog": {
+      "infile": "CHANGELOG.md"
     }
   },
   "git": {
-    "commit": false,
-    "tag": false,
-    "push": false
+    "requireBranch": "main",
+    "commitMessage": "chore: release v${version}",
+    "commit": true,
+    "tag": true,
+    "push": true,
+    "requireCleanWorkingDir": true
   },
-  "github": false,
+  "github": {
+    "release": true,
+    "tokenRef": "PUSH_TOKEN",
+    "comments": {
+      "submit": true,
+      "issue": ":rocket: _This issue has been resolved in v${version}. See [${releaseName}](${releaseUrl}) for release notes._",
+      "pr": ":rocket: _This pull request is included in v${version}. See [${releaseName}](${releaseUrl}) for release notes._"
+    }
+  },
   "npm": {
     "publish": true,
     "skipChecks": true,

--- a/package.json
+++ b/package.json
@@ -18,15 +18,10 @@
     "format:fix": "prettier --write src/**/*.ts",
     "dev": "tsc -w",
     "build": "tsc",
-    "changelog": "lerna-changelog",
-    "beta": "pnpm stable --preRelease=beta",
     "stable": "release-it --ci",
-    "release:major": "pnpm stable major",
-    "release:minor": "pnpm stable minor",
-    "release:patch": "pnpm stable patch",
-    "release:premajor": "pnpm beta premajor",
-    "release:preminor": "pnpm beta preminor",
-    "release:prepatch": "pnpm beta prepatch",
+    "release.patch": "pnpm release.ci patch",
+    "release.minor": "pnpm release.ci minor",
+    "release.major": "pnpm release.ci major",
     "prepare": "husky"
   },
   "keywords": [

